src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java

```diff
@@ -60,7 +61,8 @@ public void beginProcessing(String charset) {
     }
 
     @Override
-    protected void processFiltered(File file, List<String> lines) {
+    protected void processFiltered(File file, FileText fileText) {
+        final List<String> lines = fileText.getLines();
         detector.processLines(lines);
```

Comment:
please avoid usage of temporal object - List<String> lines , that is easy refactoring.